pkg/netns/netns.go

```diff
@@ -4,88 +4,302 @@
 package netns
 
 import (
-	"errors"
 	"fmt"
 	"os"
-	"os/exec"
+	"path"
 	"path/filepath"
+	"runtime"
 	"strings"
+	"sync"
 
-	"github.com/containernetworking/plugins/pkg/ns"
-	"github.com/vishvananda/netlink"
+	"golang.org/x/sys/unix"
+
+	// TODO consider dropping this dep entirely - it is only used here now
+	"github.com/vishvananda/netns"
 )
 
-// RemoveIfFromNetNSIfExists removes the given interface from the given
-// network namespace.
+const bindMountPath = "/run/netns" /* Bind mount path for named netns */
```

Comment:
Yes, this is something to hash out.

`vishvanand/netns` uses `/run/netns` for it's pins.

`ip netns xxx` uses `/var/run/netns`

I don't think it actually matters which location is used. Latter would make these discoverable by `ip netns` tooling, which is probably desirable.

Reply:
(for my benefit/further clarification) so the actual _correct_ location to bind netnses is any name subdirectory under `/run` - `/var/run` is a legacy location that is now always symlinked to `/run`.

`/run/netns` -> this is the path that `ip netns` uses, by convention.

`/run/docker/netns` -> this is the path docker uses - netnses bound here are still fully valid and usable, they just won't show up by default in tools like `ip netns` that hardcode the search path to `/run/netns` - this is intentional.

I don't think we need pins at all as previously discussed but if we do, it might be cleaner to put ours under `/run/cilium/netns`